File: boj/problem2460/Main.java
package problem2460;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;


public class Main {
    public static String solve(int[][] arr) {
        int answer = 0;
        int currCnt = 0;
        int maxCnt = -1;

        for (int i = 0; i < arr.length; i++) {
            currCnt = currCnt - arr[i][0] + arr[i][1];
            maxCnt = Math.max(currCnt, maxCnt);
        }

        answer = maxCnt;

        return Integer.toString(answer);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int[][] arr = new int[10][2];

        for (int i = 0; i < arr.length; i++) {
            st = new StringTokenizer(br.readLine());
            int offboard = Integer.parseInt(st.nextToken());
            int onboard = Integer.parseInt(st.nextToken());
            int[] nums = {offboard, onboard};
            arr[i] = nums;
        }

        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        bw.write(solve(arr));
        bw.flush();
    }
}

File: boj/problem2460/main.py
from typing import List


def solve(people_cnts: List[List[int]]) -> str:
    curr_cnt = 0
    max_cnt = -1
    for cnts in people_cnts:
        curr_cnt = curr_cnt - cnts[0] + cnts[1]
        max_cnt = max(curr_cnt, max_cnt)

    return str(max_cnt)


def main():
    people_cnts = []
    for i in range(10):
        offboard, onboard = map(int, input().split())
        people_cnts.append([offboard, onboard])

    print(solve(people_cnts))


if __name__ == "__main__":
    main()
